src/core/packages/elasticsearch/server-internal/src/elasticsearch_service.ts

```diff
@@ -138,6 +139,10 @@ export class ElasticsearchService
         getAgentsStats: agentManager.getAgentsStats.bind(agentManager),
       },
       publicBaseUrl: config.publicBaseUrl,
+      setCpsFeatureFlag: (enabled) => {
```

Comment:
A problem we'll face at implementation time is how to set the config in the elasticsearch setup phase when Kibana's already serving traffic.
From how I understand the feature to work is that enabling CPS will be dynamic. Project owners will be able to select pre-existing projects in the Cloud UI, which will then have to flip the switch in kibana.
Perhaps it's better to iron out the full flow at a later stage and only focus on the feature flag.